Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/OSIRISDiffractionReduction.py

```diff
@@ -624,17 +623,28 @@ def PyExec(self):
         result_map = result_map.combine(self._van_ws_map, divide_workspace)
 
         if len(result_map) > 1:
-            # Workspaces must be added to the ADS, as there does not yet exist
-            # a workspace list property (must be passed to merge runs by name).
-            temp_ws_names = ["__run_" + str(idx) for idx in range(len(result_map))]
-            for temp_ws_name, sample_ws in zip(temp_ws_names, result_map.values()):
-                mtd.addOrReplace(temp_ws_name, sample_ws)
-
-            # Merge the sample files into one.
-            output_ws = MergeRuns(
-                InputWorkspaces=temp_ws_names, OutputWorkspace="merged_sample_runs", StoreInADS=False, EnableLogging=False
-            )
-            delete_workspaces(temp_ws_names)
+            extracted_spectra = [self._extract_ws_spectra(ws, str(drange)) for drange, ws in result_map.items()]
+
+            have_same_spectra_count = len({len(spectra) for spectra in extracted_spectra}) == 1
+            if not have_same_spectra_count:
+                logger.warning("Cannot merge focussed workspaces with different number of spectra")
+                output_ws = [ws for ws in result_map.values()][0]
+            else:
+                # Group workspaces located at the same index
+                matched_spectra = [list(spectra) for spectra in zip(*extracted_spectra)]
+                # Merge workspaces located at the same index
+                merged_spectra = [
+                    MergeRuns(InputWorkspaces=spectra, OutputWorkspace="OSIRIS" + self._output_ws_name + f"_merged_{idx}")
```

Comment:
I don't think you should add the "OSIRIS" prefix here because it's already inside self._output_ws_name. Not particularly important because you now clean up these merged workspaces before but I noticed this yesterday. Thought I'd mention it today given that it looks like you'll need to make a change anyway to update a system test reference file.

Reply:
Ok, I'll remove it and update the tests